activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -118,7 +120,11 @@ def commit
         connection.release_savepoint(savepoint_name)
         super
         parent = connection.transaction_manager.current_transaction
-        records.each { |r| parent.add_record(r) }
+        if parent.uncommittable?
+          commit_records
```

Comment:
@arthurnn Thanks for helping out. I'll rebase this PR on your commit :+1: 

@jeremy I've pushed a new version, what do you think about `commit_transaction?` name? Or do you prefer `participates_in_records_transaction?`?